tools/sphere.c

```diff
@@ -80,3 +78,31 @@ int do_touch( string str ) {
 int do_gaze( string str ) {
 
 }
+
+/**
+ *  Function finds and returns the sloop shadow based on the
+ *  string passed.
+ *  @param {String} sloop_type : valid values either "slooper" or "slooped"
+ *  @param {Object} ob :  player object to test for shadows
+ *  @return {Object} : returns the shadow object
+ */
+
+string find_sloop_shadow( string sloop_type, object ob ) {
+    if ( sloop_type != "slooper" || sloop_type != "slooped" ) {
+        return 0;
+    }
+    if (! ob ) {
+        return 0;
+    }
+    object *list_shadows = shadow_list( ob );
+    int i;
+    for ( i = 0; i <= sizeof( list_shadows ); i += 1  ) {
+        string shadow_name = program_name( list_shadows[i] );
```

Comment:
We should actually talk about the difference between load_name() and program_name() next lecture. It's not super important because very rarely is it ever going to make a difference in your code, but understanding the difference can help you better understand inheritance, which is important.